include/libp2p/crypto/secp256k1_provider.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_CRYPTO_SECP256K1_PROVIDER_HPP
+#define LIBP2P_CRYPTO_SECP256K1_PROVIDER_HPP
+
+#include <array>
+#include <vector>
+
+#include <gsl/span>
+#include <libp2p/outcome/outcome.hpp>
+
+namespace libp2p::crypto::secp256k1 {
+
+  /**
+   * @brief Common types
+   */
+  using PrivateKey = std::array<uint8_t, 32>;
+  using PublicKey = std::array<uint8_t, 33>;
+  using Signature = std::vector<uint8_t>;
+
+  /**
+   * @struct Key pair
+   */
+  struct KeyPair {
+    PrivateKey private_key_; /**< Secp256k1 private key */
+    PublicKey public_key_;   /**< Secp256k1 public key */
+
+    /**
+     * @brief Comparing keypairs
+     * @param other - second keypair to compare
+     * @return true, if keypairs are equal
+     */
+    bool operator==(const KeyPair &other) {
+      return private_key_ == other.private_key_
+          && public_key_ == other.public_key_;
+    }
+
+    /**
+     * @brief Comparing keypairs
+     * @param other - second keypair to compare
+     * @return true, if keypairs aren't equal
+     */
+    bool operator!=(const KeyPair &other) {
+      return !operator==(other);
+    }
```

Comment:
Move implementation to .cpp file, keep declaration.

Reply:
I think create new .cpp just for three lines of trivial code is excessively.